cmux.go

```diff
@@ -103,22 +107,21 @@ func (m *cMux) Serve() error {
 			continue
 		}
 
-		go m.serve(c)
+		wg.Add(1)
+		go m.serve(c, &wg)
 	}
 }
 
-func (m *cMux) serve(c net.Conn) {
+func (m *cMux) serve(c net.Conn, wg *sync.WaitGroup) {
+	defer wg.Done()
+
 	muc := newMuxConn(c)
 	for _, sl := range m.sls {
 		for _, s := range sl.ss {
 			matched := s(muc.sniffer())
 			muc.reset()
 			if matched {
-				select {
-				case sl.l.connc <- muc:
-				case <-sl.l.donec:
-					_ = c.Close()
-				}
+				sl.l.connc <- muc
```

Comment:
Now, I understand the problem. We don't want a connection that has been read being queued in a sl.l.connc. 

However, this new model might create a new problem, that I really dread: what if serve is called for more than len(sl.l.connc) and the go-routine that actually calls sl.l.Accept() is dead.

The only option I can think of is to keep the code as it was before, and introduce the wait group. Once wg.Wait() returns, we can iterator through connections in connc and close them one-by-one. WDYT? Thanks!